CMakeLists.txt

```diff
@@ -83,24 +65,24 @@ file(GLOB G4sources
   ./plugins/CaloPreShowerSDAction.cpp
 )
 
-
-#file(GLOB headers ./ILD/include/*.h)
-
 if(DD4HEP_USE_PYROOT)
   ROOT_GENERATE_DICTIONARY(G__ILDEx ${headers} LINKDEF include/ROOT/LinkDef.h)
   list(APPEND sources G__ILDEx.cxx)
 endif()
 
-#Create target, library, rootmap, install
-
 add_dd4hep_plugin(${PackageName} SHARED ${sources})
-
 add_dd4hep_plugin(${PackageName}G4 SHARED ${G4sources})
+target_link_libraries(${PackageName}   DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers ${ROOT_LIBRARIES} ${LCIO_LIBRARIES})
+target_link_libraries(${PackageName}G4 DD4hep::DDCore DD4hep::DDRec DD4hep::DDParsers DD4hep::DDG4 ${ROOT_LIBRARIES} ${Geant4_LIBRARIES} ${LCIO_LIBRARIES})
 
-target_link_libraries(${PackageName} ${DD4hep_LIBRARIES} ${DD4hep_REDUCED_COMPONENT} ${ROOT_LIBRARIES} ${LCIO_LIBRARIES})
-
-target_link_libraries(${PackageName}G4 ${DD4hep_LIBRARIES} ${DD4hep_COMPONENT_LIBRARIES} ${ROOT_LIBRARIES} ${Geant4_LIBRARIES} ${LCIO_LIBRARIES})
+#make sure DD4hep comes before any other includes, fix problems with using DD4hep not from LCG view, doesn't matter
+#otherwise
+include_directories(BEFORE SYSTEM $<TARGET_PROPERTY:DD4hep::DDCore,INCLUDE_DIRECTORIES>)
 
+include_directories( SYSTEM
+  ${LCIO_INCLUDE_DIRS}
+  Boost::boost
+ )
```

Comment:
What is the reason for not having these includes also attached to targets?

Reply:
Do you mean why not use target_include_directories?

Reply:
Yes, exactly.

Reply:
just laziness. I moved to target_include_now.
Also, boost isn't needed at the moment.
I don't seem to need the BEFORE dd4hep anymore...